Source/DafnyCore/Backends/Rust/Dafny-compiler-rust.dfy

```diff
@@ -19,23 +19,44 @@ module RAST
   // Default Indentation
   const IND := "  "
 
+  opaque function FoldLeft<A(!new), T>(f: (A, T) --> A, init: A, xs: seq<T>): A
```

Comment:
That's what I was using before. Unfortunately its specification wasn't good enough so I had to reimplement it so that the function could be partial. I needed a partial function to prove termination as I was doing recursive calls in the foldleft accumulation function.